plasma/plasma-mvp/morevp-alice-bob.md

```diff
@@ -0,0 +1,103 @@
+MoreVP Alice-Bob Scenarios
```

Comment:
a general question: what is the treatment of multiple exact copies of a TX1 spending the same inputs

scenario is:
                TX1 is Alice -> Operator and Alice. If that is duplicated by the Operator, will Alice exit either her input or output? Wouldn't that allow Operator to exit too much? - technically, he didn't double spend, but has given themselves a lot of coins!

Reply:
a general remark: 
            In most of the cases it would probably be interesting to know "what if a certain action didn't happen" - how would the outcome be "broken" if someone doesn't follow protocol or take their due actions

Reply:
> The root chain contract only allows a single in-flight exit of any given transaction (by transaction hash)

:+1:

what do you think about the "what-ifing"?

Reply:
> "what if a certain action didn't happen" - how would the outcome be "broken" if someone doesn't follow protocol or take their due actions

I'll second @pdobacz's request here. For instance, what happens in the happy path if neither Alice nor Bob piggyback onto the exit? What happens if only 1 of them does it? Et cetera.

Reply:
Are the contracts available in `plasma-contracts` helpful in illuminating this flow?

As far as piggybacking specifically, as per my comments above I don't think it's useful to iterate each possible scenario. The exit game determines whether the transaction is valid (outputs get to exit) or invalid (inputs get to exit). Piggybacking then determines which of these inputs/outputs are unspent and should therefore be paid out.

Reply:
Perhaps we don't need to do every possible scenario but it would be helpful for a concrete example to clarify piggybacking. We're obviously having difficulty understanding it.

Could you confirm what happens, in the "happy path" case (transaction is valid), if:
- only Alice piggybacks
  - Alice and exit bond is refunded
  - _Alice exits the input?_
- only Bob piggybacks
  - Bob and exit bond is refunded
  - Bob exits the output
- both Alice and Bob piggyback
  - All bonds refunded
  - Bob exits the output

Is that understanding correct?